tests/test_item.py

```diff
@@ -108,6 +108,39 @@ def test_asset_absolute_href_no_item_self(self) -> None:
         actual_href = rel_asset.get_absolute_href()
         self.assertEqual(None, actual_href)
 
+    def test_item_field_order(self) -> None:
+        item = pystac.Item.from_file(
+            TestCases.get_path("data-files/item/sample-item.json")
+        )
+
+        expected_order = [
+            "type",
+            "stac_version",
+            "stac_extensions",
+            "id",
+            "geometry",
+            "bbox",
+            "properties",
+            "links",
+            "assets",
+            "collection",
+        ]
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            p = os.path.join(tmp_dir, "item.json")
+            item.save_object(include_self_link=False, dest_href=p)
+
+            with open(p) as f:
+                item_json = json.load(f)
+
+            # Assert the order matches the expected order
+            actual_order = list(item_json.keys())
+            self.assertEqual(
+                actual_order,
+                expected_order,
+                f"Order was {actual_order}, expected {expected_order}",
+            )
+
```

Comment:
We're only changing `to_dict`, so we could just serialize to a dict and check that? No need for a temp file.